packages/datetime/test/datePickerTests.tsx

```diff
@@ -26,6 +30,139 @@ describe("<DatePicker>", () => {
+            const blueprintLocaleUtils = {
+                ...LocaleUtils,
+                formatDay: () => "b",
+            };
+            const blueprintProps: IDatePickerProps = {
+                locale: "blueprint",
+                localeUtils: blueprintLocaleUtils,
+                modifiers: blueprintModifiers,
+            };
+
+            const dayPickerModifiers: IDatePickerModifiers = {
+                dayPicker: () => true,
+            };
+            const dayPickerLocaleUtils = {
+                ...LocaleUtils,
+                formatDay: () => "d",
+            };
+            const dayPickerProps: IDatePickerProps = {
+                locale: "dayPicker",
+                localeUtils: dayPickerLocaleUtils,
+                modifiers: dayPickerModifiers,
+            };
+
+            const wrapper = mount(<DatePicker {...blueprintProps} dayPickerProps={dayPickerProps} />);
+            const DayPicker = wrapper.find("DayPicker").first();
+            assert.equal(DayPicker.prop("locale"), blueprintProps.locale);
+            assert.equal(DayPicker.prop("localeUtils"), blueprintProps.localeUtils);
+            assert.equal(DayPicker.prop("modifiers"), blueprintProps.modifiers);
+        });
+
+        describe("event handlers", () => {
+            it("calls onMonthChange on button next click", () => {
+                const monthChangeSpy = sinon.spy();
+                const { root } = wrap(<DatePicker dayPickerProps={{ onMonthChange: monthChangeSpy }} />);
+                root
+                    .find(".DayPicker-NavButton--next")
+                    .first()
+                    .simulate("click");
+                assert.isTrue(monthChangeSpy.called);
+            });
+
+            it("calls onMonthChange on button prev click", () => {
+                const monthChangeSpy = sinon.spy();
+                const { root } = wrap(<DatePicker dayPickerProps={{ onMonthChange: monthChangeSpy }} />);
+                root
+                    .find(".DayPicker-NavButton--prev")
+                    .first()
+                    .simulate("click");
+                assert.isTrue(monthChangeSpy.called);
+            });
+
+            it("calls onMonthChange on month select change", () => {
+                const monthChangeSpy = sinon.spy();
+                const { root } = wrap(<DatePicker dayPickerProps={{ onMonthChange: monthChangeSpy }} />);
+                root
+                    .find({ className: Classes.DATEPICKER_MONTH_SELECT })
+                    .first()
+                    .simulate("change");
+                assert.isTrue(monthChangeSpy.called);
+            });
+
+            it("calls onDayClick", () => {
+                const mouseClickSpy = sinon.spy();
+                const { getDay } = wrap(<DatePicker dayPickerProps={{ onDayClick: mouseClickSpy }} />);
+                getDay().simulate("click");
+                assert.isTrue(mouseClickSpy.called);
+            });
+        });
+    });
```

Comment:
Add test for `calls onMonthChange on year select change` if indeed we add that functionality.